cdap-api/src/main/java/co/cask/cdap/api/dataset/ExploreProperties.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Useful class to configure dataset properties for Explore.
+ */
+@SuppressWarnings({"unused", "WeakerAccess"})
+public class ExploreProperties {
+
+  public static final String PROPERTY_EXPLORE_TABLE_NAME = "explore.table.name";
+  public static final String PROPERTY_EXPLORE_DATABASE_NAME = "explore.database.name";
+
+  /**
+   * @return the name of the Hive table configured by the properties, or null if not configured.
+   */
+  @Nullable
+  public static String getExploreTableName(DatasetProperties props) {
```

Comment:
we purposely use 'explore' instead of 'hive' to hide the implementation. For example, if we want to use Impala instead of Hive, then getHiveTableName would become wrong.